aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -357,6 +357,81 @@ Tensor mv_batching_rule(const Tensor& self, const Tensor& other) {
 }
 
+Tensor dot_batching_rule(const Tensor& self, const Tensor& other) {
+  auto self_batched = isBatchedTensor(self);
+  auto other_batched = isBatchedTensor(other);
+
+  TORCH_CHECK(/*logical*/self.dim() == 1 && /*logical*/other.dim() == 1,
+      "dot(self, other): Shape mismatch: vector "
+      "(got `self` of size ", self.sizes(), ") ",
+      "and vector (got `other` of size ", other.sizes(), ")");
+
+  // See Note [Batching rules for matmul-like operators] for why we have cases
+  if (self_batched && !other_batched) {
+    // self_physical: [..., K], other_physical: [K]
+    // View the tensors as [..., 1, K] and [K], perform matmul, and unsqueeze.
+    auto self_physical = MultiBatchVmapTransform::logicalToPhysical(self);
+    auto result = at::matmul(self_physical.tensor().unsqueeze(-2), other);
+    return self_physical.newLogicalFromPhysical(result.squeeze(-1));
+  }
+  if (!self_batched && other_batched) {
+    // self_physical: [K], other_physical: [..., K]
+    // View the tensors as [K] and [..., K, 1], perform matmul, and unsqueeze.
+    auto other_physical = MultiBatchVmapTransform::logicalToPhysical(other);
+    auto result = at::matmul(self, other_physical.tensor().unsqueeze(-1));
+    return other_physical.newLogicalFromPhysical(result.squeeze(-1));
+  }
+  if (self_batched && other_batched) {
+    // self_physical: [..., K], other_physical: [..., K]
+    // View the tensors as [..., 1, K] and [..., K, 1], perform matmul, and unsqueeze.
+    auto physical_args = MultiBatchVmapTransform::logicalToPhysical({self, other});
+    auto result = at::matmul(
+        physical_args[0].tensor().unsqueeze(-2),
+        physical_args[1].tensor().unsqueeze(-1));
+    return physical_args[0].newLogicalFromPhysical(result.squeeze(-1).squeeze(-1));
+  }
+  TORCH_INTERNAL_ASSERT(false, "either self or other must be a BatchedTensor");
+}
+
+Tensor bmm_batching_rule(const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(/*logical*/self.dim() == 3 && /*logical*/other.dim() == 3,
+      "bmm(self, other): Shape mismatch: expected 3D `self` "
+      "(got `self` of size ", self.sizes(), ") ",
+      "and 3D `other` (got `other` of size ", other.sizes(), ")");
+
+  auto physical_args = BroadcastingVmapTransform::logicalToPhysical({self, other});
+  auto result = at::matmul(physical_args[0].tensor(), physical_args[1].tensor());
+  return physical_args[0].newLogicalFromPhysical(result);
+}
+
+Tensor mm_batching_rule(const Tensor& self, const Tensor& other) {
+  auto self_batched = isBatchedTensor(self);
+  auto other_batched = isBatchedTensor(other);
+
+  TORCH_CHECK(/*logical*/self.dim() == 2 && /*logical*/other.dim() == 2,
+      "mm(self, other): Shape mismatch: expected matrix "
+      "(got `self` of size ", self.sizes(), ") ",
+      "and matrix (got `other` of size ", other.sizes(), ")");
+
+  // See Note [Batching rules for matmul-like operators] for why we have cases
```

Comment:
Soooooo... where's the Note?

Reply:
In the last PR. It's at like line 317 of this (above the definition of `mv_batching_rule`)